src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -624,6 +624,21 @@ private void RecordPresenceOfBadAttributes(ImmutableArray<CSharpAttributeData> b
             }
         }
 
+        /// <summary>Binds attributes without decoding known attributes (assumes this has already been done).</summary>
+        /// <param name="binderOpt">Binder to use. If null, <see cref="DeclaringCompilation"/> GetBinderFactory will be used.</param>
+        protected BoundAttribute[] BindAttributes(OneOrMany<SyntaxList<AttributeListSyntax>> attributeDeclarationSyntaxLists, Binder binderOpt)
```

Comment:
>BindAttributes [](http://example.com/codeflow?start=35&length=14)

It feels like this method has a lot of unnecessary overhead. Allocates a bunch of arrays, attribute data instances, etc. It also relies on helpers that are optimized towards regular attributes binding and exposing them on symbols, therefore, they are likely to have "hidden" side effects and behaviors that are not desirable or necessary for our specific scenario. Perhaps we should consider binding attributes in a more direct way, for example like SemanticModel does that. Also, we might want to make this helper less general, i.e. not expose it directly on ```Symbol``` and restrict its possible usage to our specific scenario. Then we can make an assumption that we have ```binderOpt```, for example. Lambda and local function parameters always have it. Right?

Reply:
Right, thanks. I have simplified attribute binding: 076be8220f4b65b221918b7e666ca5e3393e786c